utils/src/main/java/software/amazon/awssdk/utils/DateUtils.java

```diff
@@ -48,6 +49,18 @@ public final class DateUtils {
             .toFormatter()
             .withZone(UTC);
 
+    /**
+     * RFC 822 date/time formatter.
+     */
+    static final DateTimeFormatter RFC_822_DATE_TIME = new DateTimeFormatterBuilder()
+        .parseCaseInsensitive()
+        .parseLenient()
+        .appendPattern("EEE, dd MMM yyyy HH:mm:ss")
+        .appendLiteral(' ')
+        .appendOffset("+HHMM", "GMT")
+        .toFormatter()
+        .withLocale(Locale.US);
```

Comment:
- I guess the reason for Locale.US is [V1-refernce](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/util/DateUtils.java#L67)
- Why have we not doing 
```
toFormatter(ResolverStyle.SMART, IsoChronology.INSTANCE)
```
same as done in [RFC_1123_DATE_TIME](https://github.com/openjdk/jdk8u-dev/blob/master/jdk/src/share/classes/java/time/format/DateTimeFormatter.java#L1264) ?

Reply:
I guess an alternative would be to set the `ResolverStye.SMART` and `IsoChonology.INSTANCE` values on the DateTimeFormatter object after it's been built. Let me know if you'd prefer me to make that change.

Reply:
Yeap , we can set it after its been built.
Can we add some test cases where we can test this too (ResolverStye and IsoChonology)?